scripts/asm-installer/install_asm

```diff
@@ -1977,6 +1987,60 @@ EOF
   outro
 }
 
+pref_len() {
+  local LEN; LEN="${#1}";
+  if ((${#2} > LEN)); then
+    LEN="${#2}"
+  fi
+
+  for ((i=0; i<LEN; ++i)); do
+    if [[ "${1:i:1}" != "${2:i:1}" ]]; then
+      echo "${i}"
+      return
+    fi
+  done
+
+  echo "${LEN}"
+}
+
+find_best_region() {
+  REGION="global"
+  BEST=0
+  for location in $(gcloud beta logging locations list --format='value(location_id)'); do
+    SCORE="$(pref_len "${CLUSTER_LOCATION}" "${location}")"
+    if (( SCORE > BEST )); then
+      BEST="${SCORE}"
+      REGION="${location}"
+    fi
+  done
+
+  echo "${REGION}"
+}
+
+managed_logging_bucket_exists() {
+  BUCKET_LOCATION="$(find_best_region)"
+  FULL_LOGS_BUCKET_ID="projects/${PROJECT_ID}/locations/$(find_best_region)/buckets/${LOGS_BUCKET}"
+  BUCKET_COUNT=$(gcloud beta logging buckets list --project="$PROJECT_ID" --location="$BUCKET_LOCATION" --filter="name:${FULL_LOGS_BUCKET_ID}" --format=json | jq length)
+
+  if [[ -z "${BUCKET_COUNT}" ]] || [[ "${BUCKET_COUNT}" = '0' ]]; then
+    false
+  fi
+}
+
+establish_managed_logging_bucket() {
+  if managed_logging_bucket_exists; then return; fi
+
+  info "creating new logs bucket for managed control plane..."
+  run gcloud beta logging buckets create "${LOGS_BUCKET}" --project="${PROJECT_ID}" --location="$(find_best_region)"
```

Comment:
This requires `logging.admin` role which I think should be added to required_iam_roles list